pkg/labels/labels.go

```diff
@@ -35,8 +40,29 @@ func (s selectorOp) hasValue(val string) bool {
 	return false
 }
 
-func (s *selectorOp) match(labels Labels) bool {
+func (s *selectorOp) containsValue(val string) bool {
+	for i := range s.values {
+		if strings.Contains(val, s.values[i]) {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *selectorOp) match(labels Labels, ns string) bool {
 	val, exists := labels[s.key]
+	if s.key == k8sPodNamespace {
+		switch s.operator {
+		case opExists:
+			return s.containsValue(ns)
+		case opNotIn:
+			return !s.hasValue(ns)
+		case opIn:
+			return s.hasValue(ns)
+		default:
+			return false
+		}
+	}
```

Comment:
Here, we duplicate code which I'd rather avoid.

Reply:
thanks , I tried to merge the switch into 1 . Not sure how much more to optimize the code since there is separate handling for both cases